docs/proposals/storage-capacity-proposal.md

```diff
@@ -0,0 +1,132 @@
+pod can be scheduled at the same cycle
+
+Disadvantages:
+- Storage Capacity objects require more memory in etcd. In contrast, ACR are deleted after volume creation
+- The feature doesn't solve races issues, two pods still may reserve same disk
+
+Another option is to implement custom scheduler, which can increase performance even more and provide
+a way for users to describe storage load requirements more precise, for example set disk affinity/anti-affinity requirement for application.
+
+## Compatibility
+
+ACR can be used with CSIStorageCapacity. Scheduler uses CSIStorageCapacities at the filter stage and filters out all nodes that have not enough capacity. Custom extension points work only after default ones, so 
+using CSIStorageCapacity with ACR may decrease number of requests to extender and number of nodes in requests.
+
+## Implementation
+
+To enable the feature, controller must expose `GET_CAPACITY` capability which requires changes in `ControllerGetCapabilities` function. 
+Secondly, supported Kubernetes version should be bumped to such that supports storage capacity. The feature is stable since Kubernetes v1.24.
+To enable producing CSIStorageCapacity objects such steps are required:
+- `POD_NAME` and `NAMESPACE` environment variables for external-provisioner must be set
+like this
+```
+env:
+- name: NAMESPACE
+  valueFrom:
+    fieldRef:
+      fieldPath: metadata.namespace
+- name: POD_NAME
+  valueFrom:
+    fieldRef:
+      fieldPath: metadata.name
+```
+- Pass `--enable-capacity=true` to external-provisioner.
+- To enable ownership for CSIStorageCapacity objects and guarantee deleting objects when the driver is uninstalled, `--capacity-ownerref-level=2` argument must be passed to external-provisioner. The level indicates the number of objects that need to be traversed starting from the pod to reach the owning object for CSIStorageCapacity objects, which is `Deployment` now.
+- Add `storage.k8s.io` RBAC group
+- Following RBAC rules must be added to controller-rbac
+```
+  # Permissions for CSIStorageCapacity are only needed enabling the publishing
+  # of storage capacity information.
+  - apiGroups: ["storage.k8s.io"]
+    resources: ["csistoragecapacities"]
+    verbs: ["get", "list", "watch", "create", "update", "patch", "delete"]
+  # The GET permissions below are needed for walking up the ownership chain
+  # for CSIStorageCapacity. They are sufficient for deployment via
+  # StatefulSet (only needs to get Pod) and Deployment (needs to get
+  # Pod and then ReplicaSet to find the Deployment).
+  - apiGroups: [""]
+    resources: ["pods"]
+    verbs: ["get"]
+  - apiGroups: ["apps"]
+    resources: ["replicasets"]
+    verbs: ["get"]
+```
+To enable using these object during scheduling `storageCapacity: true` must be added to CSIDriver spec.
```

Comment:
Is this field mutable? Could user enable/disable this feature without deleting CSI?

Reply:
This field was immutable in Kubernetes <= 1.22 and mutable since 1.23 

https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.24/#csidriver-v1-storage-k8s-io

Reply:
Added information about field's mutability